packages/devtools_app/lib/src/screens/memory/panes/control/secondary_controls.dart

```diff
@@ -198,30 +195,18 @@ class _SecondaryControlsState extends State<SecondaryControls>
       children: [
         const MemorySourceDropdown(),
         const SizedBox(width: defaultSpacing),
-        if (controller.isConnectedDeviceAndroid ||
-            controller.isOfflineAndAndroidData)
-          ToggleAdbMemoryButton(
-            isAndroidCollection: widget.isAndroidCollection,
-          ),
         const SizedBox(width: denseSpacing),
-        ValueListenableBuilder<bool>(
-          valueListenable: controller.advancedSettingsEnabled,
-          builder: (context, paused, _) {
-            return controller.isAdvancedSettingsVisible
-                ? Row(
-                    children: [
-                      IconLabelButton(
-                        onPressed: controller.isGcing ? null : _gc,
-                        icon: Icons.delete,
-                        label: 'GC',
-                        minScreenWidthForTextBeforeScaling:
-                            primaryControlsMinVerboseWidth,
-                      ),
-                      const SizedBox(width: denseSpacing),
-                    ],
-                  )
-                : const SizedBox();
-          },
+        Row(
+          children: [
+            IconLabelButton(
+              onPressed: controller.isGcing ? null : _gc,
+              icon: Icons.delete,
+              label: 'GC',
+              minScreenWidthForTextBeforeScaling:
+                  primaryControlsMinVerboseWidth,
+            ),
+            const SizedBox(width: denseSpacing),
```

Comment:
pull these two children out into the parent row and delete this Row widget. Unnecessary nesting.